src/MessagePack.GeneratorCore/CodeAnalysis/Definitions.cs

```diff
@@ -182,12 +182,18 @@ public string GetDeserializeMethodString()
             }
             else if (this.primitiveTypes.Contains(this.Type))
             {
-                string suffix = this.Type == "byte[]" ? "?.ToArray()" : string.Empty;
-                return $"reader.Read{this.ShortTypeName!.Replace("[]", "s")}()" + suffix;
+                if (this.Type == "byte[]")
+                {
+                    return $"reader.ReadByteArray()";
+                }
+                else
+                {
+                    return $"reader.Read{this.ShortTypeName!.Replace("[]", "s")}()";
+                }
```

Comment:
As this seems to be the impetus for your adding of the `MessagePackReader.ReadByteArray()` method, and given my concerns about that new method that I comment on elsewhere, did you consider replacing this dependency on a `using` directive with the non-extension method syntax?

```cs
ReadOnlySequence<byte>? s = reader.ReadBytes();
byte[] b = s.HasValue ? global::System.Buffers.BuffersExtensions.ToArray(s.Value) : null;
```

Reply:
This change is made in `string GetDeserializeMethodString()`.
This method's return value is treated as `expression`.
https://github.com/neuecc/MessagePack-CSharp/blob/4b0cbcafdb621b03b05170aafa530ec21d6a8ba7/src/MessagePack.GeneratorCore/Generator/FormatterTemplate.tt#L107
The return value must not be statement(s).
This is why new api is necessary.